internal/controller/operatorconfigmap_controller.go

```diff
@@ -810,84 +796,74 @@ func (c *OperatorConfigMapReconciler) reconcileNoobaaOperatorSubscription() erro
 	return nil
 }
 
-func (c *OperatorConfigMapReconciler) reconcileCSIAddonsOperatorSubscription() error {
-	addonsSubscription, err := c.getSubscriptionByPackageName("odf-csi-addons-operator")
-	if kerrors.IsNotFound(err) {
-		addonsSubscription, err = c.getSubscriptionByPackageName("csi-addons")
-	}
+func (c *OperatorConfigMapReconciler) reconcileWebhookService() error {
+	svc := &corev1.Service{}
+	svc.Name = templates.WebhookServiceName
+	svc.Namespace = c.OperatorNamespace
+	err := c.createOrUpdate(svc, func() error {
+		if err := c.own(svc); err != nil {
+			return err
+		}
+		utils.AddAnnotation(svc, "service.beta.openshift.io/serving-cert-secret-name", "webhook-cert-secret")
+		templates.WebhookService.Spec.DeepCopyInto(&svc.Spec)
+		return nil
+	})
 	if err != nil {
 		return err
 	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != addonsSubscription.Spec.Channel {
-		addonsSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(addonsSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'odf-csi-addons-operator' to %v: %v", c.subscriptionChannel, err)
-		}
-	}
+	c.log.Info("successfully reconcile webhook service")
 	return nil
 }
 
-func (c *OperatorConfigMapReconciler) reconcileCephCSIOperatorSubscription() error {
-	cephCsiOperatorSubscription, err := c.getSubscriptionByPackageName("cephcsi-operator")
-	if kerrors.IsNotFound(err) {
-		cephCsiOperatorSubscription, err = c.getSubscriptionByPackageName("ceph-csi-operator")
-	}
+func (c *OperatorConfigMapReconciler) reconcileSubscriptionLabelForWebhook() error {
+	sub, err := c.getSubscriptionByPackageName("ocs-client-operator")
 	if err != nil {
 		return err
 	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != cephCsiOperatorSubscription.Spec.Channel {
-		cephCsiOperatorSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(cephCsiOperatorSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'cephcsi-operator' to %v: %v", c.subscriptionChannel, err)
+	if utils.AddLabel(sub, subscriptionLabelKey, subscriptionLabelValue) {
+		if err := c.update(sub); err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
-func (c *OperatorConfigMapReconciler) reconcileRecipeOperatorSubscription() error {
-	recipeOperatorSubscription, err := c.getSubscriptionByPackageName("recipe")
-	if err != nil {
-		return err
-	}
-	if c.subscriptionChannel != "" && c.subscriptionChannel != recipeOperatorSubscription.Spec.Channel {
-		recipeOperatorSubscription.Spec.Channel = c.subscriptionChannel
-		if err := c.update(recipeOperatorSubscription); err != nil {
-			return fmt.Errorf("failed to update subscription channel of 'recipe' to %v: %v", c.subscriptionChannel, err)
+func (c *OperatorConfigMapReconciler) reconcileSubscriptions() error {
+	installPlansForApproval := map[string]struct{}{}
+
+	for _, pkgName := range managedPackageNames {
```

Comment:
You should not treat all subscription the same, each one might have its own nuances, even if they are not manifested today. Please make sure each subscription is dealt with its own reconcile function

Also, as a general rule, one reconcile function need to only deal with one output resource

Reply:
done.